sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala

```diff
@@ -462,4 +463,27 @@ class OrcQuerySuite extends QueryTest with BeforeAndAfterAll with OrcTest {
       }
     }
   }
+
+  test("column nullability and comment - write and then read") {
+    val schema = StructType(
+      StructField("cl1", IntegerType, nullable = false,
+        new MetadataBuilder().putString("comment", "test").build()) ::
```

Comment:
Yeah I am afraid it is. I just grepped through the code base and there are a few places where we do this, for example:
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L1434-L1438
https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala#L397-L404

+1 for adding a convenience method.